qemu/tests/qemu_img_lock_reject_boot.py

```diff
@@ -0,0 +1,83 @@
+import logging
+
+from virttest import utils_misc
+
+from qemu.tests.qemu_disk_img import QemuImgTest
+from qemu.tests.qemu_disk_img import generate_base_snapshot_pair
+
+
+def verify_img_locked_ouput(test, output, lock_msg, img_file=None):
+    """
+    Verify QEMU image lock output.
+
+    :param test: test object
+    :param output: QEMU actual output string
+    :param lock_msg: expected lock msgs string
+    :param img_file: expected img_file name string
+    """
+    lock_msgs = lock_msg.split(";")
+    if img_file:
+        lock_msgs.append(img_file)
+    if not all(msg in output for msg in lock_msgs):
+        test.fail("Image lock information is not as expected.")
+
+
+def run(test, params, env):
+    """
+    QEMU img lcok reject boot tests.
+
+    1. create an external snapshot based on a os image (optional)
+    2. boot one vm from the os image
+    3  boot another vm from the same base os image or the snapshot
+    4. verify qemu image lock
+
+    :param test: Kvm test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment
+    """
+    def _create_os_snapshot():
+        """Crate one external snapshot based on the os image."""
+        logging.info("Create a qcow2 snapshot based on the os image.")
+        # workaround to asign syetem disk's image_name to image_name_image1
+        params["image_name_image1"] = params["image_name"]
+        gen = generate_base_snapshot_pair(params["image_chain"])
+        _, snapshot = next(gen)
+        QemuImgTest(test, params, env, snapshot).create_snapshot()
+
+    # img_file is used for checking qemu output
+    # delete once BZ1626436 is fixed
+    is_qcow2_test = params["image_format"] == "qcow2"
+    img_file = params["image_name"] if is_qcow2_test else None
+
+    if params.get("create_snapshot", "no") == "yes":
+        _create_os_snapshot()
+        vm1 = env.get_vm("avocado-vt-vm1")
+        # remove sn in 'images' to prevent boot with two images
+        vm1.params["images"] = "image1"
+    else:
+        vm1 = env.get_vm("avocado-vt-vm1")
+    vm2 = vm1.clone(name=params["second_vm_name"])
+
+    logging.info("Boot one vm from the base os image.")
+    vm1.create()
```

Comment:
Set 'start_vm = yes' in the configuration will be easy to boot the image than what you implement. Then create snapshot and boot snapshot, or boot the same image. You can decide how to achieve this. Just comment it to let you know.

Reply:
I knew, the difference is in this case it will create the sn first then boot vm1 up.
If `start_vm = yes`, vm1 will boot up first then create the sn.

Reply:
Yes, you are right. Both of them are allowed and there is almost no difference.